File: legacy_perl/gc_content.pl
# Original Perl script by I. Petrov (RU) 2012
#!/usr/bin/env perl
# legacy script: compute GC content per sequence in a FASTA file
use strict;
use warnings;
use Bio::SeqIO;
my $in = shift or die "Usage: $0 sequences.fasta\n";
my $seqio = Bio::SeqIO->new(-file => $in, -format => 'fasta');
while (my $seq = $seqio->next_seq) {
    my $id = $seq->display_id;
    my $seqstr = $seq->seq;
    my $gc = (($seqstr =~ tr/GCgc//) / length($seqstr)) * 100;
    printf "%s\t%.2f\n", $id, $gc;
}

File: biopython_rewrite/gc_content.py
# Migrated from BioPerl to Biopython by A. Smith (USA) 2020
#!/usr/bin/env python3
# Biopython rewrite: compute GC content per sequence
from Bio import SeqIO
import sys
if len(sys.argv) < 2:
    print('Usage: gc_content.py input.fasta'); sys.exit(1)
for rec in SeqIO.parse(sys.argv[1], 'fasta'):
    seq = str(rec.seq)
    gc = (seq.upper().count('G') + seq.upper().count('C')) / len(seq) * 100 if len(seq)>0 else 0
    print(f"{rec.id}\t{gc:.2f}")
